Add Test Annotation when Compiling to Java (#2395)
dafny-lang/dafny: RELEASE_NOTES.md

> https://github.com/Dargones/libraries# Upcoming

Reviewer: ```suggestion
# Upcoming
```

Rogue copy-and-paste perhaps? :)

Author: Thanks for catching that, @robin-aws.  Has been adjusted.

Reviewer: Thanks for fixing them! If you resolve the latest conflicts from mainline this should be ready to merge.

Author: Done!

- feat: Command-line arguments are now available from `Main` in Dafny programs, using `Main(args: seq<string>)` (https://github.com/dafny-lang/dafny/pull/2594)
- feat: generate warning when 'old' has no effect (https://github.com/dafny-lang/dafny/pull/2610)
- fix: Missing related position in failing precondition (https://github.com/dafny-lang/dafny/pull/2658)
- fix: No more IDE crashing on the elephant operator (https://github.com/dafny-lang/dafny/pull/2668)

> - feat: Methods and function methods marked with test attr will receive JUnit annotation when compiled to Java (https://github.com/dafny-lang/dafny/pull/2395)

Reviewer: ```suggestion
- feat: Methods and function methods marked with the `{:test}` attribute will receive a JUnit annotation when compiled to Java (https://github.com/dafny-lang/dafny/pull/2395)
```

# 3.7.1

- fix: The Dafny runtime library for C# is now compatible with .NET Standard 2.1, as it was before 3.7.0. Its version has been updated to 1.2.0 to reflect this. (https://github.com/dafny-lang/dafny/pull/2277)

> - feat: Methods and function methods marked with test attr will receive JUnit annotation when compiled to Java (https://github.com/dafny-lang/dafny/pull/2395)

Reviewer 2: Please move this up to "upcoming" section.

Author: Done; note has been moved to upcoming section.
